src/useMouse.ts

```diff
@@ -0,0 +1,63 @@
+import {useState, useEffect, useRef, RefObject} from 'react';
+
+export interface State {
+  docX: number;
+  docY: number;
+  posX: number;
+  posY: number;
+  elX: number;
+  elY: number;
+  elH: number;
+  elW: number;
+}
+
+const useMouse = (ref: RefObject<HTMLElement>): State => {
+  const frame = useRef(0);
+  const [state, setState] = useState<State>({
+    docX: 0,
+    docY: 0,
+    posX: 0,
+    posY: 0,
+    elX: 0,
+    elY: 0,
+    elH: 0,
+    elW: 0,
+  });
+
+  useEffect(() => {
+    const handler = (event: MouseEvent) => {
+      frame.current = requestAnimationFrame(() => {
```

Comment:
Should there be 

```js
cancelAnimationFrame(frame.current)
```

before this line?

Reply:
Yeah, I think so. I noticed I didn't put in above [this line](https://github.com/streamich/react-use/blob/master/src/useScroll.ts#L19) in `useScroll` either 😱